Test failure `test_normalfloat`

[Test here](https://github.com/automl/ConfigSpace/runs/4953440972?check_suite_focus=true#step:4:949) when built from wheel.
Python: `Python 3.7`

Edit: Also [here](https://github.com/automl/ConfigSpace/runs/4953441119?check_suite_focus=true#step:4:949)
Python: `Python 3.9`

Edit: It seems to be a difference in `x86_64` and `i686`, at least past the 14th decimal.


```python
  _____________________ TestHyperparameters.test_normalfloat _____________________
  
  self = <test_hyperparameters.TestHyperparameters testMethod=test_normalfloat>
  
      def test_normalfloat(self):
          # TODO test non-equality
          f1 = NormalFloatHyperparameter("param", 0.5, 10.5)
          f1_ = NormalFloatHyperparameter("param", 0.5, 10.5)
          self.assertEqual(f1, f1_)
          self.assertEqual(
              "param, Type: NormalFloat, Mu: 0.5 Sigma: 10.5, Default: 0.5",
              str(f1))
          self.assertEqual(f1.get_neighbors(0.5, rs=np.random.RandomState(42)),
                           [5.715498606617943, -0.9517751622974389, 7.300729650057271,
  >                         16.491813492284265])
  E       AssertionError: Lists differ: [5.715498606617943, -0.9517751622974389, 7.3007296500572725, 16.49181349228427] != [5.715498606617943, -0.9517751622974389, 7.300729650057271, 16.491813492284265]
  E       
  E       First differing element 2:
  E       7.3007296500572725
  E       7.300729650057271
  E       
  E       - [5.715498606617943, -0.9517751622974389, 7.3007296500572725, 16.49181349228427]
  E       ?                                                          ^^                  ^
  E       
  E       + [5.715498606617943, -0.9517751622974389, 7.300729650057271, 16.491813492284265]
  E       ?                                                          ^                  ^^
  
  /project/test/test_hyperparameters.py:209: AssertionError
```

--- a/test/test_hyperparameters.py
+++ b/test/test_hyperparameters.py
@@ -205,9 +205,8 @@ class TestHyperparameters(unittest.TestCase):
             "param, Type: NormalFloat, Mu: 0.5 Sigma: 10.5, Default: 0.5",
             str(f1))
 
-        # Due to seemingly different numbers when built with a wheel, we check that it
-        # is almost equal
-        # Recieved theses numbers, where last two are slightly different
+        # Due to seemingly different numbers with x86_64 and i686 architectures
+        # we got these numbers, where last two are slightly different
         # [
         #   5.715498606617943, -0.9517751622974389,
         #   7.3007296500572725, 16.49181349228427

--- a/test/test_hyperparameters.py
+++ b/test/test_hyperparameters.py
@@ -204,9 +204,24 @@ class TestHyperparameters(unittest.TestCase):
         self.assertEqual(
             "param, Type: NormalFloat, Mu: 0.5 Sigma: 10.5, Default: 0.5",
             str(f1))
-        self.assertEqual(f1.get_neighbors(0.5, rs=np.random.RandomState(42)),
-                         [5.715498606617943, -0.9517751622974389, 7.300729650057271,
-                          16.491813492284265])
+
+        # Due to seemingly different numbers when built with a wheel, we check that it
+        # is almost equal
+        # Recieved theses numbers, where last two are slightly different
+        # [
+        #   5.715498606617943, -0.9517751622974389,
+        #   7.3007296500572725, 16.49181349228427
+        # ]
+        # They are equal up to 14 decimal places
+        expected = [
+            5.715498606617943, -0.9517751622974389,
+            7.300729650057271, 16.491813492284265
+        ]
+        np.testing.assert_almost_equal(
+            f1.get_neighbors(0.5, rs=np.random.RandomState(42)),
+            expected,
+            decimal=14
+         )
 
         # Test attributes are accessible
         self.assertEqual(f1.name, "param")

